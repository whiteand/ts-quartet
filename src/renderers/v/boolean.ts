import { ISchemaRenderer } from "../../types";
import { getSimpleCondition } from "./utils";

export const booleanRenderer: ISchemaRenderer<null> = getSimpleCondition(
  valueId => `typeof ${valueId} === 'boolean'`,
  valueId => `typeof ${valueId} !== 'boolean'`
);
