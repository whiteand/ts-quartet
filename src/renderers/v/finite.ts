import { ISchemaRenderer } from '../../types'
import { getSimpleCondition } from './utils'

export const finiteRenderer: ISchemaRenderer<null> = getSimpleCondition(
  valueId => `Number.isFinite(${valueId})`,
  valueId => `!Number.isFinite(${valueId})`,
)
