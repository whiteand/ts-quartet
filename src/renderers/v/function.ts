import { ISchemaRenderer } from "../../types";
import { getSimpleCondition } from "./utils";

export const functionRenderer: ISchemaRenderer<null> = getSimpleCondition(
  valueId => `typeof ${valueId} === 'function'`,
  valueId => `typeof ${valueId} !== 'function'`
);
