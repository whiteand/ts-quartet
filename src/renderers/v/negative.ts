import { ISchemaRenderer } from '../../types'
import { getSimpleCondition } from './utils'

export const negativeRenderer: ISchemaRenderer<null> = getSimpleCondition(
  valueId => `${valueId} < 0`,
  valueId => `!(${valueId} < 0)`,
)
