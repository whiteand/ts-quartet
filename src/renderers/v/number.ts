import { getSimpleCondition } from "./utils";

const nanRenderer: SchemaRenderer<number> = getSimpleCondition(
  (valueId) => `Number.isNaN(${valueId})`,
  (valueId) => `!Number.isNaN(${valueId})`
);

export const numberRenderer: SchemaRenderer<number> = {
  getExpr: (schema, valueId, ctxId, mapCtx) => {},
  getNotExpr: (schema, valueId, ctxId, mapCtx) => {},
  getIfExprReturnTrue: (schema, valueId, ctxId, mapCtx) => {},
  getIfNotExprReturnFalse: (schema, valueId, ctxId, mapCtx) => {},
};
