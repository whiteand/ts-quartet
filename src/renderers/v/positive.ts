import { ISchemaRenderer } from "../../types";
import { getSimpleCondition } from "./utils";

export const positiveRenderer: ISchemaRenderer<null> = getSimpleCondition(
  valueId => `${valueId} > 0`,
  valueId => `!(${valueId} > 0)`
);
