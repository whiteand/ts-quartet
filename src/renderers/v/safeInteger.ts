import { getSimpleCondition } from './utils'

export const safeIntegerRenderer = getSimpleCondition<null>(
  valueId => `Number.isSafeInteger(${valueId})`,
  valueId => `!Number.isSafeInteger(${valueId})`,
)
