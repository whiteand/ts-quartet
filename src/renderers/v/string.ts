import { ISchemaRenderer } from "../../types";
import { getSimpleCondition } from "./utils";

export const stringRenderer: ISchemaRenderer<null> = getSimpleCondition(
  valueId => `typeof ${valueId} === 'string'`,
  valueId => `typeof ${valueId} !== 'string'`
);
