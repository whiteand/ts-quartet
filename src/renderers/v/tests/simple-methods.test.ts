import { booleanRenderer } from '../boolean'
import { finiteRenderer } from '../finite'
import { functionRenderer } from '../function'
import { numberRenderer } from '../number'
import { stringRenderer } from '../string'
import { symbolRenderer } from '../symbol'
import { positiveRenderer } from '../positive'
import { negativeRenderer } from '../negative'
import { safeIntegerRenderer } from '../safeInteger'
import { snapshot } from './snapshot'

describe('simple methods', () => {
  test('v.boolean', () => {
    snapshot(booleanRenderer, { schema: null })
  })
  test('v.finite', () => {
    snapshot(finiteRenderer, { schema: null })
  })
  test('v.function', () => {
    snapshot(functionRenderer, { schema: null })
  })
  test('v.number', () => {
    snapshot(numberRenderer, { schema: null })
  })
  test('v.string', () => {
    snapshot(stringRenderer, { schema: null })
  })
  test('v.symbol', () => {
    snapshot(symbolRenderer, { schema: null })
  })
  test('v.positive', () => {
    snapshot(positiveRenderer, { schema: null })
  })
  test('v.negative', () => {
    snapshot(negativeRenderer, { schema: null })
  })
  test('v.safeInteger', () => {
    snapshot(safeIntegerRenderer, { schema: null })
  })
})
