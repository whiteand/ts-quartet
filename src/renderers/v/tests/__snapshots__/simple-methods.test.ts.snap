// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`simple methods v.boolean 1`] = `"typeof _valueId === 'boolean'"`;

exports[`simple methods v.boolean 2`] = `"typeof _valueId !== 'boolean'"`;

exports[`simple methods v.boolean 3`] = `"if (typeof _valueId === 'boolean') return true;"`;

exports[`simple methods v.boolean 4`] = `"if (typeof _valueId !== 'boolean') return false;"`;

exports[`simple methods v.finite 1`] = `"Number.isFinite(_valueId)"`;

exports[`simple methods v.finite 2`] = `"!Number.isFinite(_valueId)"`;

exports[`simple methods v.finite 3`] = `"if (Number.isFinite(_valueId)) return true;"`;

exports[`simple methods v.finite 4`] = `"if (!Number.isFinite(_valueId)) return false;"`;

exports[`simple methods v.function 1`] = `"typeof _valueId === 'function'"`;

exports[`simple methods v.function 2`] = `"typeof _valueId !== 'function'"`;

exports[`simple methods v.function 3`] = `"if (typeof _valueId === 'function') return true;"`;

exports[`simple methods v.function 4`] = `"if (typeof _valueId !== 'function') return false;"`;

exports[`simple methods v.negative 1`] = `"_valueId < 0"`;

exports[`simple methods v.negative 2`] = `"!(_valueId < 0)"`;

exports[`simple methods v.negative 3`] = `"if (_valueId < 0) return true;"`;

exports[`simple methods v.negative 4`] = `"if (!(_valueId < 0)) return false;"`;

exports[`simple methods v.number 1`] = `"typeof _valueId === 'number'"`;

exports[`simple methods v.number 2`] = `"typeof _valueId !== 'number'"`;

exports[`simple methods v.number 3`] = `"if (typeof _valueId === 'number') return true;"`;

exports[`simple methods v.number 4`] = `"if (typeof _valueId !== 'number') return false;"`;

exports[`simple methods v.positive 1`] = `"_valueId > 0"`;

exports[`simple methods v.positive 2`] = `"!(_valueId > 0)"`;

exports[`simple methods v.positive 3`] = `"if (_valueId > 0) return true;"`;

exports[`simple methods v.positive 4`] = `"if (!(_valueId > 0)) return false;"`;

exports[`simple methods v.safeInteger 1`] = `"Number.isSafeInteger(_valueId)"`;

exports[`simple methods v.safeInteger 2`] = `"!Number.isSafeInteger(_valueId)"`;

exports[`simple methods v.safeInteger 3`] = `"if (Number.isSafeInteger(_valueId)) return true;"`;

exports[`simple methods v.safeInteger 4`] = `"if (!Number.isSafeInteger(_valueId)) return false;"`;

exports[`simple methods v.string 1`] = `"typeof _valueId === 'string'"`;

exports[`simple methods v.string 2`] = `"typeof _valueId !== 'string'"`;

exports[`simple methods v.string 3`] = `"if (typeof _valueId === 'string') return true;"`;

exports[`simple methods v.string 4`] = `"if (typeof _valueId !== 'string') return false;"`;

exports[`simple methods v.symbol 1`] = `"typeof _valueId === 'symbol'"`;

exports[`simple methods v.symbol 2`] = `"typeof _valueId !== 'symbol'"`;

exports[`simple methods v.symbol 3`] = `"if (typeof _valueId === 'symbol') return true;"`;

exports[`simple methods v.symbol 4`] = `"if (typeof _valueId !== 'symbol') return false;"`;
